app/src/main/java/org/eyeseetea/malariacare/data/repositories/UserAccountRepository.java

```diff
@@ -1,125 +1,40 @@
- *  it under the terms of the GNU General Public License as published by
- *  the Free Software Foundation, either version 3 of the License, or
- *  (at your option) any later version.
- *
- *  Health Network QIS App is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- *  GNU General Public License for more details.
- *
- *  You should have received a copy of the GNU General Public License
- *  along with Foobar.  If not, see <http://www.gnu.org/licenses/>.
- */
-
 package org.eyeseetea.malariacare.data.repositories;
 
-import android.content.Context;
-
-import org.eyeseetea.malariacare.data.IDataSourceCallback;
 import org.eyeseetea.malariacare.data.IUserAccountDataSource;
-import org.eyeseetea.malariacare.data.database.datasources.UserAccountLocalDataSource;
-import org.eyeseetea.malariacare.data.database.utils.Session;
-import org.eyeseetea.malariacare.data.remote.sdk.UserAccountDhisSDKDataSource;
-import org.eyeseetea.malariacare.domain.boundary.IRepositoryCallback;
-import org.eyeseetea.malariacare.domain.boundary.repositories.IUserAccountRepository;
-import org.eyeseetea.malariacare.domain.entity.Credentials;
+import org.eyeseetea.malariacare.data.NetworkStrategy;
+import org.eyeseetea.malariacare.domain.boundary.IUserAccountRepository;
 import org.eyeseetea.malariacare.domain.entity.UserAccount;
+import org.eyeseetea.malariacare.domain.usecase.UserFilter;
 
 public class UserAccountRepository implements IUserAccountRepository {
-    IUserAccountDataSource userAccountLocalDataSource;
-    IUserAccountDataSource userAccountRemoteDataSource;
-
-    public UserAccountRepository(Context context) {
 
-        userAccountLocalDataSource = new UserAccountLocalDataSource(context);
-        userAccountRemoteDataSource = new UserAccountDhisSDKDataSource(context);
-    }
+    private final IUserAccountDataSource userAccountAPIDataSource;
+    private final IUserAccountDataSource userAccountLocalDataSource;
 
-    @Override
-    public void login(final Credentials credentials,
-            final IRepositoryCallback<UserAccount> callback) {
-        if (credentials.isDemoCredentials()) {
-            localLogin(credentials, callback);
-        } else {
-            remoteLogin(credentials, callback);
-        }
+    public UserAccountRepository(IUserAccountDataSource userAccountAPIDataSource,
+                                 IUserAccountDataSource userAccountLocalDataSource){
+        this.userAccountAPIDataSource = userAccountAPIDataSource;
+        this.userAccountLocalDataSource = userAccountLocalDataSource;
     }
 
     @Override
-    public void logout(final IRepositoryCallback<Void> callback) {
+    public UserAccount getUser(UserFilter userFilter, NetworkStrategy networkStrategy) {
+        UserAccount userAccount = userAccountLocalDataSource.getUser(userFilter);
```

Comment:
if NetworkStrategy is NetworkFirst always is invoked userAccountLocalDataSource, this one only should be executed if NetworkStrategy is LocalFirst or if NetworkFirst call fail